sdk/core/ts-http-runtime/review/ts-http-runtime.api.md

```diff
@@ -450,7 +446,7 @@ export interface TlsSettings {
 
 // @public
 export interface TokenCredential {
-    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    getToken(scopes: string | string[]): Promise<AccessToken | null>;
```

Comment:
Wouldn't we want some form of cancellation if getting a Token is taking too long?

Reply:
@mpodwysocki Won't that process of cancellation be different from provider to provider? Should we leave that to the user instead?

Reply:
I think Matt is referring to client-side request cancellation not operation-on-server cancellation.

Reply:
I see so this is different than `abort`? or is this analogous to `request timeout`?

Reply:
The ask is to pass an AbortSignal in the method options and listen to the abort event on it, and if triggered, abort the request.

Reply:
shameless plug: this util could be useful to address this: https://github.com/Azure/azure-sdk-for-js/blob/d47f5b3befcb3201da4a79e136bbfcc7bf520bfc/sdk/core/core-util/src/createAbortablePromise.ts#L21-L27